pkg/ring/util.go

```diff
@@ -25,5 +33,56 @@ func GenerateTokens(numTokens int, takenTokens []uint32) []uint32 {
 		tokens = append(tokens, candidate)
 		i++
 	}
+
+	// Ensure generated tokens are sorted.
+	sort.Slice(tokens, func(i, j int) bool {
+		return tokens[i] < tokens[j]
+	})
+
```

Comment:
Given this is not an hot path, I added a sorting here as well to simplify testing, otherwise we would have to sort tokens in some tests. Thoughts?

Reply:
It isn't clear that this sorting was added for tests, especially since other non-test calls do sorting afterwards. Perhaps `GeneratedSortedTokens` in test code would be a better place to put sorting to?

Reply:
Yes, done.